### ISSUE TITLE
Prisma Studio or Data Browser: `Do not know how to serialize a BigInt`

### ISSUE THREAD

**Issue by yovanoc** [prisma/studio#614]
On the latest version

<img width="1461" alt="image" src="https://user-images.githubusercontent.com/7302317/105364739-cbdc5480-5bfd-11eb-97f2-104e1a551333.png">

**Comment by KhaVNguyen** [comment]
@yovanoc I am getting this exact error as well when trying to view my data inside Prisma Studio (some of my columns are BigInt's).

This can be reproduced by just having any model with any field that is a`BigInt`

**Comment by KhaVNguyen** [comment]
I tried to fix it using the proposed solution here in my own code: 
![image](https://user-images.githubusercontent.com/12026775/105438621-369a8980-5c18-11eb-8585-1c95fee4bf84.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt

I believe it needs to be done on Prisma's end though

**Comment by yoroshikun** [comment]
I am also experiencing this issue with `BigInt`.
I hope this can be solved soon. This seems to affect prisma as a whole as well

**Comment by madebysid** [comment]
Hey everyone, I'm going to take a look at this very soon, I can reproduce the problem as well.

@yoroshikun I can't seem to be able to see an issue with Prisma Client or Migrate on 2.15.0, could you elaborate what you mean by ` This seems to affect prisma as a whole as well`? Thanks!

**Comment by yoroshikun** [comment]
Sorry for the late reply, I have dome some testing and the issue is not from Prisma not knowing how to deserialize but the framework next.js not being able to. (It affected both pages and API routes). 

But yes as described above I can also reproduce the issue with Prisma studio and for now, have swapped to storing the values as strings as a fallback.

**Comment by madebysid** [comment]
Hello everyone! So this should be fixed in Prisma version [2.16.0](https://github.com/prisma/prisma/releases/tag/2.16.0) (due today). Please let me know if you're still seeing issues after it goes out. 

@yoroshikun Prisma Client returns real JS `BigInts` for Prisma `BigInts`, which aren't serializable, maybe that's why NextJS is also struggling.

**Comment by tsdexter** [comment]
@madebysid I am getting this error on an introspected wordpress database running on Microsoft SQL Server (azure) with Yoast (wordpress-seo) installed. The call I'm using is is in a simple `express` server:

```
app.get("/select", async (req, res) => {
  try {
    const table = req.query.table;
    const limit = parseInt(req.query.limit, 10);
    const data = await prisma[table].findMany({
      take: limit,
    });
    res.send(data);
  } catch (error) {
    res.status = 500;
    res.send({ error: true, message: error.message });
  }
});
```
On the route `http://localhost:3001/select?table=wp_yoast_seo_links&limit=10` the response is: 

```
{
  "error": true,
  "message": "Do not know how to serialize a BigInt"
}
```

How can I go about returning data with `BigInt`s in it?

**Comment by tsdexter** [comment]
@madebysid I've fixed this for now by changing the schema like so: 

```
model wp_yoast_seo_links {
  // id             BigInt  @id @default(autoincrement())
  id             Int     @id @default(autoincrement())
  url            String  @db.NVarChar(255)
  // post_id        BigInt
  post_id        Int
  // target_post_id BigInt
  target_post_id Int
  type           String  @db.NVarChar(8)
  language       String? @db.NVarChar(32)
  region         String? @db.NVarChar(32)

  @@index([post_id, type], name: "link_direction")
}
```

All of the `BigInt`'s in my schema are for `id` and foreign key `id` fields on third party tables...whereas the first party tables such as `wp_posts` just use regular `Int` so I would imagine I shouldn't have any issues with this change - can you think of any possible issues with this? Thanks

**Comment by madebysid** [comment]
Hey @tsdexter, IIUC Studio is not involved in your setup at all, right? Is this is general question? (Happy to help either way, just wondering)

The underlying issue is that when you use a `BigInt` in a field in your schema, not only does your DB use a `BigInt` (if it is supported), but Prisma Client also now returns a real Javascript [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) for that field. Weirdly enough, the native `JSON.stringify` does not work with BigInts. If you try running `JSON.stringify(BigInt(34))` in a Node REPL, you'll see this error. It doesn't work in browser's either.

So in your case, the issue is that you're using `res.send` in your request handler, and express internally tries to `JSON.stringify` your response: [res.send in express](https://github.com/expressjs/express/blob/master/lib/response.js#L158), [express' stringify function](https://github.com/expressjs/express/blob/master/lib/response.js#L1118)

Again, this has nothing to do with Express, it's just that these two JS language features do not work together for whatever reason.

So here are a few ways to address this:

1. You could change the `BigInt` to an `Int` like you've already done. This is far from ideal, since you might actually WANT a `BigInt`.
2. Treat `BigInt`s as strings. In your specific case, you might be able to get away with a `BigInt().toString()` to convert the integer into a string. That way, Express will see your `BigInt` as a string, and `JSON.stringify` will work as expected. It'll be manual work to make sure you change all `BigInt`s to strings, but it might be feasible, depending on what your (client-side) app wants.
3. Take care of stringifying`BigInt`s yourself. This is a lot more involved (and is ultimately what I do in Studio as well). Here's the gist of it: https://dev.to/benlesh/bigint-and-json-stringify-json-parse-2m8p. This is very similar to 2., but it works better if you want real `BigInt`s client-side. If you prefix all BigInt strings with a prefix, you can identify such strings and run `BigInt("prefix::33434".slice(0, 8))` client side to get back a real `BigInt`. Just be careful that you use a prefix that won't also naturally appear in regular strings.

Hopefully this was helpful and gave you a few ideas! if you have more questions, it'd probably be more useful to the community if you could start a [discussion](https://github.com/prisma/prisma/discussions/new), and tag me there, I'm happy to help.

**Comment by ryands17** [comment]
@tsdexter 👋 

The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.

```ts
BigInt.prototype.toJSON = function() {       
  return this.toString()
}
```

**Comment by snettah** [comment]
@ryands17 good catch, but for me bigint are string in an object, I've used this instead.

```javascript
BigInt.prototype.toJSON = function () {
  return Number(this);
};

**Comment by ryands17** [comment]
@snettah 👋 

This will work fine if the number is in the limits that JavaScript supports. For very large numbers, you would need to use a string.

**Comment by waptik** [comment]
> @tsdexter 👋
> 
> The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.
> 
> ```ts
> BigInt.prototype.toJSON = function() {       
>   return this.toString()
> }
> ```


Hi! I am using prisma client 3.5.0 with nextjs v12.0.4 and i am facing this same issue.
Where do i implement your solution?
I tried it in a `*.d.ts` file but it didn't have any effect

**Comment by Vrishtrix** [comment]
> > @tsdexter 👋
> > The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.
> > ```ts
> > BigInt.prototype.toJSON = function() {       
> >   return this.toString()
> > }
> > ```
> 
> Hi! I am using prisma client 3.5.0 with nextjs v12.0.4 and i am facing this same issue. Where do i implement your solution? I tried it in a `*.d.ts` file but it didn't have any effect

I used this as a workaround.

```ts
const data = await prisma.data.findMany();
const updatedData = JSON.stringify(data, (_key, value) => {
    typeof value === 'bigint' ? value = value.toString() : value
})

return { props: { updatedData } }
```

**Comment by roscadalexandru** [comment]
this is my solution 
`BigInt.prototype.toJSON = function () {
  const int = Number.parseInt(this.toString());
  return int ?? this.toString();
};`

**Comment by necmettin** [comment]
> @tsdexter 👋
> 
> The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.
> 
> ```ts
> BigInt.prototype.toJSON = function() {       
>   return this.toString()
> }
> ```

Where do you add this?

**Comment by maheshsuthar93** [comment]
> @tsdexter 👋
> 
> The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.
> 
> ```ts
> BigInt.prototype.toJSON = function() {       
>   return this.toString()
> }
> ```

where to add this one could send more code here?

**Comment by pedrosc94** [comment]
Any update on this, I'm new at using prisma, and my auto generated ids on cockroachdb are BigInt, how to i solve the problem? If i change to Int in schema it kinda solves the problem, but if I do a GET request the given number wont work unless i mannualy change the value on the database to a smaller number

**Comment by AndreLucas2103** [comment]
> > @tsdexter 👋
> > A solução mais fácil a partir de agora é adicionar o `toJSON`método no protótipo para que a serialização funcione bem.
> > ```ts
> > BigInt . protótipo . toJSON  =  function ( )  {        
> >   return  this . toString ( ) 
> > }
> > ```
> 
> onde adicionar este você poderia enviar mais código aqui?

Crie um arquivo patch.js na raiz de seu projeto e importe ele para o server

**Comment by AndreLucas2103** [comment]
> > @tsdexter 👋
> > A solução mais fácil a partir de agora é adicionar o `toJSON`método no protótipo para que a serialização funcione bem.
> > ```ts
> > BigInt . protótipo . toJSON  =  function ( )  {        
> >   return  this . toString ( ) 
> > }
> > ```
> 
> onde adicionar este você poderia enviar mais código aqui?

Crie um arquivo patch.js contendo
`BigInt.prototype.toJSON = function() {
    return this.toString()
} `

logo após importe ele para sua fonte
`import './patch.js'; or require('patch.js')
`

**Comment by frankps** [comment]
Had the same issue. All above was not working, but this worked (added to src/main.ts):

(BigInt.prototype as any).toJSON = function () {
  return this.toString();
};

**Comment by JayP127** [comment]
SkSoju, the same problem here zkSoju. I don't know how or where to handle the suggested implementation. Anybody from Prisma can shed some light? Thanks!

**Comment by zkSoju** [comment]
> SkSoju, the same problem here zkSoju. I don't know how or where to handle the suggested implementation. Anybody from Prisma can shed some light? Thanks!
> 
> I rather use the official solution for handling BigInt; I only need a more thorough explanation from Prisma

Agreed that would probably be ideal, currently using strings, but it's a bit annoying to keep track of. If anyone has any examples that would be very helpful.

**Comment by zkSoju** [comment]
> SkSoju, the same problem here zkSoju. I don't know how or where to handle the suggested implementation. Anybody from Prisma can shed some light? Thanks!
> 
> I rather use the official solution for handling BigInt; I only need a more thorough explanation from Prisma

I'm using the official implementation in each of the service files and it works. Ideally the earlier solution could be used so you don't have to use the custom implementation for every query.

**Comment by luanvdw** [comment]
Hey @zkSoju, can you share the version of Prisma you are using in the project where you are experiencing the `Do not know how to serialize a BigInt` error?

**Comment by janpio** [comment]
Are you testing Prisma Studio @leafyshark? If not, please report this at https://github.com/prisma/prisma and include more information. We would be very interesting in figuring this out of course. (Are you using raw queries in that test? Please include reply in the new issue. Thanks.)

**Comment by gspasov** [comment]
> @tsdexter 👋
> 
> The easiest workaround as of now is to add the `toJSON` method on the prototype so that serialisation works well.
> 
> ```ts
> BigInt.prototype.toJSON = function() {       
>   return this.toString()
> }
> ```

If you are using TypeScript and/or ESLint, you'll need to add few more lines to make TypeScript and/or ESLint happy. The comment lines may vary depending on how you have setup your ESLint.
```ts
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment      <-- Necessary for my ESLint setup
  // @ts-ignore: Unreachable code error                              <-- BigInt does not have `toJSON` method
  BigInt.prototype.toJSON = function (): string {
    return this.toString();
  };
```

On the question of _"Where to add this statement?"_ 
Add it to the root of your project. In my case that would be the `index.ts` where I initialize my server which uses Prisma as ORM.

**Comment by Krillegeddon** [comment]
Wherever I used JSON.serialize() in my project, I sprinkled this code next to my real class...

`export class MyClass {
    // My method, doing stuff... like
    //var mySerializeString = JSON.serialize(myObject);
}

// Some bug with Prisma and BigInt... https://github.com/prisma/studio/issues/614
(BigInt.prototype as any).toJSON = function () {
    return Number(this)
};
`
And it seems to work. Reading a one and a half year old thread, and the solution is this - not good.

**Comment by petradonka** [comment]
I agree, having to use this workaround after this much time is not good enough.

We've been looking into this internally and have fixed a few cases that could lead to this error. We have a few more to go though, and reproduction is a challenge in some cases.

If you have more info on how to reproduce the issue in Studio (or Data Browser in Prisma Data Platform), we'd like to hear from you, to better understand where things go wrong and how you're using `BigInt`s.

**Comment by Krillegeddon** [comment]
I tried console.log(result) after a read of bigints. Serialization worked, but I noticed that all values are printed as 1n, 2n, 3n etc.

However, when trying to do console.log(JSON.stringify(result)) - I get the error.

I have no idea why JSON should serialize bigints differently than normal ints? I think it ought to be the same, but just possibly there are more digits? But, I just tried that, and the number is truncated and exponent is shown...

However, problem seems to be in JSON and not Prisma. Prisma is building the object by the book it seems.

However, I can see there is a package called json-bigint. If they have managed to do a workaround that makes JSON work by just including a package, perhaps you can make that package a dependency to Prisma? Or implement the same thing? I haven't tried it though, I am happy with my current workaround...

**Comment by Rainer2465** [comment]
@Krillegeddon is right, the problem is with JSON itself and not prisma, JSON does not support BigInt.
`JSON.stringify({a:10n})` will throw the same exception: 'Uncaught TypeError: Do not know how to serialize a BigInt'

btw: the 'n' suffix means it's a bigInt

I've found this article: https://dev.to/benlesh/bigint-and-json-stringify-json-parse-2m8p it helped me to make a work around

```
// Serialize
const json = JSON.stringify(data, (key, value) =>
  typeof value === "bigint" ? `BIGINT::${value}` : value
);

// Deserialize
const backAgain = JSON.parse(json, (key, value) => {
  if (typeof value === "string" && value.startsWith('BIGINT::')) {
    return BigInt(value.substr(8));
  }
  return value;
});
```

**Comment by Rainer2465** [comment]
> @Rainer2465 @Krillegeddon @gspasov you have some nice ideas on fixing this locally for you, that's great! We would really appreciate if you give us some hints on where the error happens for you(**studio/data browser**) as well as any other useful information that would help us reproduce this bug locally(**prisma version, schema of the model where the error happens, any other steps you found that get you to this error**) Thanks a lot for helping us out!

Anywhere on the browser when you try to serialize a BigInt on JSON, for me for example all my models that have BigInt when I try so send it to the frontend using Express it throws this error. You can reproduce this error on your browsers console with JSON.stringify({a:10n})

**Comment by petradonka** [comment]
@Rainer2465, are you running into issues with Studio/Data Browser when using/serializing BigInts, or is this a more generic Prisma client issue for you?

**Comment by alexanderattar** [comment]
Any updates on this? Why is the issue closed? It still appears to be an issue

**Comment by petradonka** [comment]
We haven't been able to reproduce the issue and get to the bottom of it. @alexanderattar, are you consistently running into this? Is it in Studio or the Data Browser? Could you share a bit more details, such as how you're using `BigInt`s and which steps you're taking that result in the error?

**Comment by kcmv** [comment]
Hi @petradonka, I am also experiencing this issue using the Browser but in Prisma Studio there's no issue.

**Comment by petradonka** [comment]
Thanks for reaching out @kcmv. Are you consistently running into this? Could you share a bit more details, such as how you're using BigInts and which steps you're taking that result in the error?
Which Prisma Studio version are you using?

**Comment by mirelalazar** [comment]
Jumping in here, just got this error by simply running the following raw query:
let sql = `SELECT count(*) as counter FROM Users`
const data: any[] = await prisma.$queryRaw(Prisma.raw(sql))

**Comment by mcgrealife** [comment]
For anyone encountering this issue in next.js, I found this article helpful https://flaviocopes.com/nextjs-serialize-date-json/ (unrelated to Prisma)

**Comment by mcgrealife** [comment]
> For anyone encountering this issue in next.js, I found this article helpful https://flaviocopes.com/nextjs-serialize-date-json/ (unrelated to Prisma)

However, this technique requires type casting the parsed and stringifyed prisma response as a prisma type. So the linked technique is a nextjs quick-fix only.

**Comment by petradonka** [comment]
Thanks for the details @kcmv!
We've also made some progress on this issue and will be releasing some improvements with the next Prisma release that should fix the issue in most cases.

**Comment by jboothe** [comment]
If you have made it this far in the thread and have found none of the suggestions above worked, you may want to try what finally worked for me. The solutions above actually do work, but there was a cacheing issue in my app that prevented any of the changes to be compiled. 

Simply enough all it took was to delete the `/dist` folder and restart `npm run start:dev`. 

In my NestJs app I found this solution works in my main.ts file:

**Comment by EnkhAmar** [comment]
> If you have made it this far in the thread and have found none of the suggestions above worked, you may want to try what finally worked for me. In my case, there was a cacheing issue in my app that prevented new changes from being compiled.
> 
> Simply enough all it took was to delete the `/dist` folder and restart `npm run start:dev`. Both solutions below then worked for me. Choose one or the other, or you may want to implement @Rainer2465 serialize/deserialize option if your case requires it.
> 
> ## Global solution
> In my NestJs app, [this solution](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) works in my main.ts file:
> 
> ```
> // main.ts
>   // eslint-disable-next-line @typescript-eslint/ban-ts-comment      <-- Necessary for my ESLint setup
>   // @ts-ignore: Unreachable code error                              <-- BigInt does not have `toJSON` method
>   BigInt.prototype.toJSON = function (): string {
>     return this.toString();
>   };
> ```
> 
> ## Service level solution
> And the suggested [Prisma solution](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields#serializing-bigint) works in each service method:
> 
> ```
> // some.service.ts
> async findAll() {
>     const data = await this.prisma.caseType.findMany();
>     return JSON.stringify(data, (key, value) =>
>       typeof value === 'bigint' ? value.toString() : value,
>     );
> }
> ```

This works me for like a charm.

**Comment by trevorr** [comment]
No need to fight with eslint in Typescript:

```ts
declare global {
  interface BigInt {
    toJSON(): string;
  }
}

BigInt.prototype.toJSON = function (): string {
  return this.toString();
};
```

**Comment by lucasvalentee** [comment]
If you're trying to use this with Typescript, I suggest you create a .ts or .js file with:

```typescript
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore: Unreachable code error
BigInt.prototype.toJSON = function (): number {
  return this.toString();
};
```

You can also return Number(this), but I suggest toString as it has no character limiter. However, it will depend on your context.

After that, you must import or require this file in your main file (like index.ts/.js or main.ts, etc).

If you are trying to run tests with Jest, you must add this file in the installation configuration. These settings can be created in package.json with "jest: { .... }" or in a file called jest.config.ts (create this in the root).

Example:

```typescript
export default {
  testTimeout: 30000,
  moduleFileExtensions: ['js', 'json', 'ts'],
  **setupFiles: ['./my_config_file'],**
  testRegex: '.*\\.spec\\.ts$',
  transform: {
    '^.+\\.(t|j)s$': 'ts-jest',
  },
  collectCoverageFrom: ['**/*.(t|j)s'],
  coverageDirectory: '../coverage',
  testEnvironment: 'node',
}
```

At this point you can create a test module, with many configs, and create an index file that imports all the configs, and just pass this index file in the setupFiles config from jest.

**Comment by lucasvalentee** [comment]
> Our tests are failing due to this. Not sure why, they were working until we upgraded to 4.0.0
> 
> ![Screenshot 2022-06-29 at 22 33 59](https://user-images.githubusercontent.com/27443324/176548967-c033a3df-0d80-4fa8-aab0-b932e92d0fdd.png)

This should help you: https://github.com/prisma/studio/issues/614#issuecomment-1466901384

**Comment by phoenixeliot** [comment]
I believe v4 started returning a lot more non-Number types (BigInt, Decimal, maybe others). Honestly a huge pain, we're looking for a good workaround too. The worst part is that if you get a Decimal back, + does string concatenation instead of actual addition, so it broke a lot of our backend endpoints and made them produce wrong results. We have to cast every $queryRaw count/sum/etc result with Number() now.

**Comment by GaleanoWilliams** [comment]
Y en serverless framework, dondé pondríamos ?

`
BigInt.prototype.toJSON = function (): number {
  return this.toString();
};
`

**Comment by lucasvalentee** [comment]
> Y en serverless framework, dondé pondríamos ?
> 
> `BigInt.prototype.toJSON = function (): number { return this.toString(); };`

Maybe on handler or main file. Example:

Create a file like bigint.ts with: BigInt.prototype.toJSON = function (): string (the return type is string here because toString didnt returns a number) { return this.toString(); }

And on the main file from your project, you should import this file like: "import '${PATH_TO_FILE}/bigint';

I hope this help you

**Comment by davialcantaraa** [comment]
What worked for me:

```typescript
// types/common.d.ts

declare global {
  interface BigInt {
    toJSON(): string
  }
}
``` 

```typescript
// somewhere.ts

// ...

BigInt.prototype.toJSON = function () {
    return String(this)
}

// ...
}
```

**Comment by Noitidart** [comment]
Just a note, as I see some confusion above, where some think this is due to Next.js or other things. I got this error after upgrading from Prisma 3 to Prisma 4. So it's in Prisma 4.

**Comment by angelobordin** [comment]
In my Prisma schema i am using INT for some properties, in the normal functions like "findMany" the results came perfect, But when I use QueryRaw the result are BigInt.
I resolve this mistake using this loop function to convert the results. This is a temporary.
`public async getFormList(userParams: RequestUserParams, prismaService: OmitPrismaClient) {
		try {
			const result = await prismaService.$queryRaw<
				supplier_form[]
			>`SELECT f.* FROM supplier_form f LEFT JOIN supplier_form r ON f.id = r.revision_of_form_id WHERE r.id IS NULL OR f.revision_version = (SELECT MAX(revision_version) FROM supplier_form WHERE id = f.id OR revision_of_form_id = f.id)ORDER BY f.id;`;

			const resultFixed = result.map((form) => {
				let convertedObj = {};
				for (const key in form) {
					if (Object.prototype.hasOwnProperty.call(form, key)) {
						const value = form[key];

						// Verificar se o valor é BigInt e converter para Number
						convertedObj[key] = typeof value === "bigint" ? Number(value) : value;
					}
				}

				return convertedObj;
			});

			return resultFixed;
		} catch (error) {
			throw error;
		}
	}`

**Comment by 456wyc** [comment]
This problem seems to still exist, and we hope that we can convert BigInt into String directly in the query result, which is what TypeOrm does.

**Comment by 456wyc** [comment]
> This problem seems to still exist, and we hope that we can convert BigInt into String directly in the query result, which is what TypeOrm does.

The bigint field is the first in the prisma query result.
In typeorm it is the second:
![image](https://github.com/prisma/studio/assets/3086528/a8f9a30f-f9fa-43c4-939a-fad95858d37e)

**Comment by TechGunners** [comment]
having the same issue but with a different msg: JSON.stringify cannot serialize BigInt.
But the interesting part is I'm nowhere using JSON.stringify 

```typescript

			return db.blog.findMany({
				where: { published: true },
				take: 500,
				select: {
					id: true,
					title: true,
					cover: true,
					views: true,
					created: true,
					user: { select: { username: true, name: true, pp: true, verified: true } }
				}
			})
```

**Comment by angelopedroso** [comment]
This worked for me

```ts
return { ...table, id: Number(table.id) }
```

**Comment by Sir-Aguiar** [comment]
I was having the same issue and here's what worked for me.

In my index.ts file where my script's started i defined this piece of code:

```ts
declare global {
  interface BigInt {
    toJSON(): string;
  }
}

BigInt.prototype.toJSON = function () {
  return this.toString();
};
```

After that all my `BigInt` instances are converted to string automatically when the method `toJSON()` is called. I didn't have to make any changes in my code besides that. So, if i have and array with objects that have some fields of type BigInt, all of them become strings.
In my HTTP response, for example, i receive every bigint type as a string and then i convert them back to BigInt

**Comment by igalklebanov** [comment]
Hey :wave:

Prisma ORM 7 introduces a brand new standalone [Studio](https://www.prisma.io/docs/orm/tools/prisma-studio), built from the ground up to work everywhere, with or without Prisma ORM. Consequently, all "old Studio" issues are being closed as there won't be any further work on it. :mega:

The new Studio is also available in the Prisma VS Code extension and in the Prisma Console application. :electric_plug:

Help shape, improve and fix the new Studio with us in this repository by submitting bug reports and feature requests. :construction_worker:
